File: app/src/main/java/com/arjanvlek/oxygenupdater/internal/ViewHelper.java
package com.arjanvlek.oxygenupdater.internal;

import android.view.View;

/**
 * @author Adhiraj Singh Chauhan (github.com/adhirajsinghchauhan)
 */
public final class ViewHelper {
	public static void clear(View v) {
		v.setAlpha(1);
		v.setScaleY(1);
		v.setScaleX(1);
		v.setTranslationY(0);
		v.setTranslationX(0);
		v.setRotation(0);
		v.setRotationY(0);
		v.setRotationX(0);
		v.setPivotY(v.getMeasuredHeight() >> 2);
		v.setPivotX(v.getMeasuredWidth() >> 2);
		v.animate().setInterpolator(null).setStartDelay(0);
	}
}

File: app/src/main/java/com/arjanvlek/oxygenupdater/internal/ViewHelper.kt
package com.arjanvlek.oxygenupdater.internal

import android.view.View

/**
 * @author Adhiraj Singh Chauhan (github.com/adhirajsinghchauhan)
 */
object ViewHelper {

    fun clear(view: View) {
        view.alpha = 1f

        view.scaleY = 1f
        view.scaleX = 1f

        view.translationY = 0f
        view.translationX = 0f

        view.rotation = 0f
        view.rotationY = 0f
        view.rotationX = 0f

        view.pivotY = (view.measuredHeight shr 2.toFloat().toInt()).toFloat()
        view.pivotX = (view.measuredWidth shr 2.toFloat().toInt()).toFloat()

        view.animate().setInterpolator(null).startDelay = 0
    }
}
